sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamReader.scala

```diff
@@ -311,6 +311,37 @@ final class DataStreamReader private[sql](sparkSession: SparkSession) extends Lo
   @Experimental
   def text(path: String): DataFrame = format("text").load(path)
 
+  /**
+   * Loads text file(s) and returns a [[Dataset]] of String. The underlying schema of the Dataset
```

Comment:
It might be weird to add var args, since the streaming case would always be to watch a directory (not list a bunch of files).  I think its fine to leave it out for now.

This is existing, but its a little odd that the methods in this file talk about `loading files` rather than `watching directories of files and processing them as they appear`.